server/cpho/jinja2/indicators/upload/preview_upload.jinja2

```diff
@@ -0,0 +1,132 @@
+{% extends 'base.jinja2' %}
+{% from 'cpho_macros.jinja2' import check_error, check_error_class %}
+{% block content %}
+
+
+ 
+  <a id='upload_btn'
+     href="#"
+     onclick="showLoading()"
+     hx-post="{{ url("save_upload") }}"
+     class="btn btn-primary {{ disabled if not no_errors }}">{{ tdt("Upload Data") }}</a>
+  <button id='loading_btn'
+          class="btn btn-primary d-none"
+          type="button"
+          disabled>
+    <span class="spinner-grow spinner-grow-sm" role="status" aria-hidden="true"></span>
+    {{ tdt("Uploading Data") }}
+  </button>
```

Comment:
Hm, few issues here.

First, there's multiple accessibility issues with "disabling" a link through this `a.disabled` class. First, its styling information doesn't do anything to communicate info to a screen reader user. Second, `pointer-events: none;` doesn't stop keyboard events, so anyone using keyboard navigation can still "click" this link. You _could_ use the `aria-disabled="true"` attribute instead, to tell screen readers it's disabled, and change the CSS selector to `a[aria-disabled="true"]`, but that doesn't fix the keyboard interaction issue.

The proper fix is to not use an `a` here at all. Only use an anchor tag if the purpose is actually navigation. Accessibility aside, there's [lots of other issues](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#onclick_events) with using an `a` with `href="#"` in place of a `button`.

Additionally, if you use a `button` instead, then all you need to do to both prevent _any_ interaction, and to provide a11y support, is to add the [`disabled` attribute ](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attributes). Bootstrap even already has styles for `button[disabled="true"]`.

The second issue is this `#loading_btn` button. This one's more minor, but a `button`, even with the `disabled` and `role="status"` attributes, shouldn't be used for communicating status. It's weird [html semantics](https://developer.mozilla.org/en-US/docs/Glossary/Semantics#semantics_in_html), so you're fighting against the browser to make it work (having to `disable` the button behaviour). Better to just use a `span` with the bootstrap button styling (the `role="status"` is good though, leave that for sure :+1: )